src/test/java/org/terasology/web/resources/console/ConsoleResourceTest.java

```diff
@@ -50,19 +86,44 @@ public void testMessageNotification() {
         verify(observer).onEvent(argThat(ResourcePath::isEmpty), eq(message), eq(clientEntity));
     }
 
+    @Test
+    public void testCommandExecutionFails() throws ResourceAccessException {
+        setupConsole();
+
+        expectedException.expect(ResourceAccessException.class);
+        consoleResource.getPostMethod(ResourcePath.createEmpty()).perform("testCommand testArg", client);
+    }
+
     @Test
     public void testCommandExecution() throws ResourceAccessException {
-        Console consoleMock = mock(Console.class);
-        Context context = new ContextImpl();
-        context.put(Console.class, consoleMock);
-        ConsoleResource consoleResource = new ConsoleResource();
+        context = new ContextImpl();
+        consoleResource = new ConsoleResource();
+        NetworkSystem networkSystemMock = mock(NetworkSystem.class);
+        when(networkSystemMock.getMode()).thenReturn(NetworkMode.NONE);
+        context.put(NetworkSystem.class, networkSystemMock);
+        Console consoleWithCommandsSpy = spy(new ConsoleImpl(context));
+        context.put(Console.class, consoleWithCommandsSpy);
         InjectionHelper.inject(consoleResource, context);
 
-        Client client = mock(Client.class);
-        EntityRef clientEntityMock = mock(EntityRef.class);
+        client = mock(Client.class);
+        clientEntityMock = mock(EntityRef.class);
+        when(clientEntityMock.getComponent(ClientComponent.class)).thenReturn(new ClientComponent());
         when(client.getEntity()).thenReturn(clientEntityMock);
 
-        consoleResource.getPostMethod(ResourcePath.createEmpty()).perform("testCommand testArg", client);
-        verify(consoleMock).execute("testCommand testArg", clientEntityMock);
+        ServerCommands serverCommands = new ServerCommands();
+        MethodCommand.registerAvailable(serverCommands, consoleWithCommandsSpy, context);
+
+        consoleResource.getPostMethod(ResourcePath.createEmpty()).perform("save", client);
+        verify(consoleWithCommandsSpy).execute("save", clientEntityMock);
+    }
+
+    @Test
+    public void testHelpCommand() throws ResourceAccessException {
+        setupConsole();
+
+        ResourceObserver resourceObserverMock = mock(ResourceObserver.class);
+        consoleResource.setObserver(resourceObserverMock);
+        consoleResource.getPostMethod(ResourcePath.createEmpty()).perform("help", client);
+        verify(consoleMock, never()).execute(anyString(), eq(clientEntityMock));
```

Comment:
I agree with verifying that no real command is actually executed by the console.

However unless I'm missing something, `resourceObserverMock` is never used. Maybe the idea was to verify that it received a notification with the help message? I think just verifying it received a notification (`any()` matchers could be useful) should be enough.

Reply:
`resourceObserverMock` is used because otherwise the `notifyEvent` method in `AbstractObservableResource` throws an NPE, since it tries to access the observer.